core/src/main/java/site/ycsb/Client.java

```diff
@@ -325,6 +324,7 @@ public static void main(String[] args) {
     }
 
     Thread terminator = null;
+    final CountDownLatch l = new CountDownLatch(1);
```

Comment:
The use of the CountDownLatch I think obscures what is it doing. I'd suggest moving to a Semaphore. 

Here would be:

```java
final Semaphore done = new Semaphore(1);
done.acquire();
```  

Below it would be:
```java
done.release();
```

Then in the hook:
```java
if( ! done.tryAquire() ) {
  // ...stopWorkload()
  try {
    if( ! done.tryAquire(5, TimeUnit.SECONDS) ) {
     System.err.println("Shutdown await timeout. Exit.");
    }
  } catch (InterruptedException timeout) {
  }
}
```

Extra credit to move this to a Phaser where things can then track the progress of the run (init, workload, cleanup, export measurement). In this case, as an example, you probably don't want to trigger the hook if we are already in the export phase.

Reply:
Done. I am sorry I didnt have time this time to port it to Phaser.